src/Analyzer/Passes/OptimizeRedundantFunctionsInOrderByPass.cpp

```diff
@@ -0,0 +1,120 @@
+#include <Analyzer/Passes/OptimizeRedundantFunctionsInOrderByPass.h>
+#include <Analyzer/ColumnNode.h>
+#include <Analyzer/FunctionNode.h>
+#include <Analyzer/InDepthQueryTreeVisitor.h>
+#include <Analyzer/QueryNode.h>
+#include <Analyzer/SortNode.h>
+#include <Functions/IFunction.h>
+
+namespace DB
+{
+
+namespace
+{
+
+class OptimizeRedundantFunctionsInOrderByVisitor : public InDepthQueryTreeVisitor<OptimizeRedundantFunctionsInOrderByVisitor>
```

Comment:
In our style guide, first we write public part, then private.

Reply:
I don't see it [there](https://clickhouse.com/docs/en/development/style)

Reply:
It is implicit style guide. Most of the classes follow it.
It is hard to find scenario where we want to see all implementation details first, before public interface.

Reply:
I'll change it, but I see no point in it. It's already an implementation detail and is hidden in the `.cpp` file.